Applications/TextEditor/TextEditor.h

```diff
@@ -0,0 +1,15 @@
+#pragma once
+
+#include <LibGUI/GApplication.h>
+#include <LibGUI/GTextEditor.h>
+#include <LibGUI/GWindow.h>
+
+class TextEditorWidget final : public GWidget {
+public:
+    TextEditorWidget(GApplication* app, GWindow* window, const int& argc, char** argv);
```

Comment:
There is only one instance of `GApplication` globally, so you don't need to pass it to `TextEditorWidget`.
Instead, it can get to the global `GApplication` by using `GApplication::the()`.

There's also no need to pass the `GWindow`, since it can just call `window()` on itself to retrieve the the same information :)

Arguments parsing should be done in `main()` so let's not pass `argc` and `argv` either.